scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -211,6 +215,32 @@ def test_dijkstra_min_only_random(n):
             p = pred[p]
 
 
+@pytest.mark.parametrize('n', (10, 100, 500))
+def test_star_graph(n):
+    # Build the star graph
+    star_arr = np.zeros((n, n), dtype=float)
+    star_center_idx = 0
+    star_arr[star_center_idx, :] = star_arr[:, star_center_idx] = range(n)
+    G = scipy.sparse.csr_matrix(star_arr, shape=(n, n))
+    # Build the distances matrix
+    SP_solution = np.zeros((n, n), dtype=float)
+    SP_solution[:] = star_arr[star_center_idx]
+    for idx in range(1, n):
+        SP_solution[idx] += star_arr[idx, star_center_idx]
+    np.fill_diagonal(SP_solution, 0)
+
+    # Run the tests
+    def check(method, directed):
+        SP = shortest_path(G, method=method, directed=directed)
+        assert_allclose(
+            SP_solution, SP
+        )
+
+    for method in methods:
+        for directed in (True, False):
```

Comment:
Could these also be parameterised?

I know it's not the standard in much of the csgraph tests, but I think we should consider going for a more modern approach from here. Plus if only 1 method fails, we know exactly what method fails

Reply:
Done.